src/app/components/add-class-form/add-class-form.component.ts

```diff
@@ -74,14 +80,30 @@ export class AddClassFormComponent implements OnInit {
     }
   }
 
-  ngOnInit(): void {
+
+  ngOnInit() {
+
     this.getTeachersToSelectData();
-    this.httpOptionsService.loadHeaders();
 
+    this.httpOptionsService.loadHeaders();
     this.form = this.formBuilder.group({
         "character":[null, [requiredValidator("character is required"), maxLengthValidator(1,"character must be 1 symbol"), patternValidator(/[A-Za-zÀ-ÿ]/,"must be single character")]],
-        "number":[null, [requiredValidator("number is required"), maxValueValidator(12, "max value is 12"), minValueValidator(1, "min value is 1")]],
-        "classTeacherId": [null, [requiredValidator("teacher is required")]]
-      }) 
+        "number":[null, [requiredValidator("number is required"), maxValueValidator(12, "max value is 12")]],
+        "classTeacherId": [null, [requiredValidator("teacher is required")]],
+        "selectedPupils": [null]
+      });
+      this.onChanges();
+  }
+  onChanges(): void {
+   this.form.get('number').valueChanges.subscribe(val => {
+      if(val != ""){
+        this.showPupils = true;
+        console.log(typeof(val));
+        this.getPupilsToSelectData(val);
+      }
+      else{
+        this.showPupils = false;
+      }
+    });
```

Comment:
I don't understand how all this code is aligned

Reply:
Ctrl+A then Ctrl+K+F